dbg-trace.sh.in

```diff
@@ -30,19 +30,19 @@ typeset _Dbg_shell='@SH_PROG@'
 
 # The short shell name. Helps keep code common in bash, zsh, and ksh debuggers.
 # Equivalent to
-typeset _Dbg_shell_name=${_Dbg_shell##*/}
+typeset _Dbg_shell_name="${_Dbg_shell##*/}"
 
 typeset -a _Dbg_script_args; _Dbg_script_args=("$@")
 
 # Original $0. Note we can't set this in an include.
-typeset _Dbg_orig_0=$0
+typeset _Dbg_orig_0="$0"
 
 # Equivalent to basename $0; the short program name
-typeset _Dbg_pname=${0##*/}
+typeset _Dbg_pname="${0##*/}"
 
 ## Stuff set by autoconf/configure ###
-typeset prefix=@prefix@
-typeset _Dbg_libdir=@PKGDATADIR@
+# @PKGDATADIR@ often uses $prefix, _Dbg_libdir must be a global variable
+_Dbg_assign_libdir() { typeset prefix="@prefix@"; _Dbg_libdir="@PKGDATADIR@"; }; _Dbg_assign_libdir
```

Comment:
Zsh's way with an anonymous function should be this

```suggestion
function { typeset prefix="@prefix@"; _Dbg_libdir="@PKGDATADIR@"; }
```

https://zsh.sourceforge.io/Doc/Release/Functions.html#Anonymous-Functions